activerecord/test/cases/encryption/performance/encryption_performance_test.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require "cases/encryption/helper"
+require "models/book"
+require "models/post"
+
+class ActiveRecord::Encryption::EncryptionPerformanceTest < ActiveRecord::TestCase
+  fixtures :encrypted_books, :posts
+
+  setup do
+    ActiveRecord::Encryption.config.support_unencrypted_data = true
+  end
+
+  test "performance when saving records" do
+    baseline = -> { create_post_without_encryption }
+
+    assert_slower_by_at_most 1.4, baseline: baseline do
+      create_post_with_encryption
+    end
```

Comment:
Is this something we should be asserting?  Performance will change depending on the environment (as we're seeing in CI), and this type of test couples the performance of the two calls.  IOW, even if the encrypted version never gets slower, we're not allowed to improve performance of the non encrypted version.

What is the big picture thing we're trying to prevent?  If creating a post with encryption is 1.5 times slower than without, why is that a failure?

Reply:
@tenderlove The goal is preventing performance regressions when modifying the library.  In my experience going with generous thresholds makes these tests stable (I haven't tuned much yet), but you are right that it couples the performance of both calls.

I can move them out of the main pipeline, maybe behind a specific `rake test` task? I think they are valuable to keep.

Reply:
I think keeping track of performance is a great idea, but I don't think it's worth failing builds.  Especially when we can't control the hardware and stability of the CI infrastructure.  I think moving them to a different rake task is a good idea.

If we could keep track of the perf over time, it would make spotting regressions easy (but that sounds like a lot of work).  A separate rake task would at least allow us to `git bisect` to find the problem.

Reply:
@tenderlove 👆addressed in https://github.com/basecamp/rails/commit/4a9a8d2da364de7b61669a2f664d5a6c8cac1e75

You can run them for different adapters like:

```bash
$ rake test:encryption:performance:mysql2
```

And they are not part of the regular build anymore.